lib/carrierwave/uploader/versions.rb

```diff
@@ -213,16 +213,24 @@ def url(*args)
       # Recreate versions and reprocess them. This can be used to recreate
       # versions if their parameters somehow have changed.
       #
-      def recreate_versions!(*versions)
+      def recreate_versions!(*names)
         # Some files could possibly not be stored on the local disk. This
         # doesn't play nicely with processing. Make sure that we're only
         # processing a cached file
         #
         # The call to store! will trigger the necessary callbacks to both
         # process this version and all sub-versions
-        if versions.any?
-          file = sanitized_file if !cached?
-          store_versions!(file, versions)
+
+        if names.any?
+          file = sanitized_file
+          # We should somehow filter versions to cache in #cache_versions!.
+          # Passing arguments through all of those callbacks is difficult, so @versions_to_cache
+          # instance variable is introduced. It's used only here and in #cache_versions!.
+          @versions_to_cache = substitute_dependent_with_parents(names)
```

Comment:
I would avoid code like this at the current moment. I would find a way to extract this or improve the current method. I know that there might be others places where we are instantiating variables outside of the constructor of the class, but as of now, I see this as a bad smell and should be avoided at all costs.